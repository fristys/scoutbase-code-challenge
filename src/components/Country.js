import React from 'react';

export function Country({ history, country }) {
  const goToCountry = code => history.push(`/countries/${code}`);

  return (
    <div className="country" onClick={() => goToCountry(country.code)}>
      <div className="code">{country.code}</div>

      <div className="name">
        {country.name} ({country.continent.name})
      </div>

      <div className="languages">
        {country.languages.map(language => `${language.native} (${language.name})`).join(', ')}
      </div>
    </div>
  );
}
