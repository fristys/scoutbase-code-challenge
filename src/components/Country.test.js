import React from 'react';
import { shallow } from 'enzyme';

import { Country } from './Country';

const mockData = {
  history: {
    push: location => location
  },

  country: {
    name: 'Andorra',
    code: 'AD',
    continent: { name: 'Europe', __typename: 'Continent' },
    languages: [
      { code: 'ca', native: 'CatalÃ ', name: 'Catalan', __typename: 'Language' }
    ],
    __typename: 'Country'
  }
};

describe('Country component', () => {
  it('Renders without crashing', () => {
    const { history, country } = mockData,
      cmp = shallow(<Country history={history} country={country} />);

    expect(cmp).toMatchSnapshot();
  });

  it('Shows country code', () => {
    const { history, country } = mockData,
      cmp = shallow(<Country history={history} country={country} />),
      code = <div className="code">{country.code}</div>;

    expect(cmp.contains(code)).toEqual(true);
  });

  it('Shows country name', () => {
    const { history, country } = mockData,
      cmp = shallow(<Country history={history} country={country} />),
      name = (
        <div className="name">
          {country.name} ({country.continent.name})
        </div>
      );

    expect(cmp.contains(name)).toEqual(true);
  });

  it('Shows country languages', () => {
    const { history, country } = mockData,
      cmp = shallow(<Country history={history} country={country} />),
      languages = (
        <div className="languages">
          {country.languages
            .map(language => `${language.native} (${language.name})`)
            .join(', ')}
        </div>
      );

    expect(cmp.contains(languages)).toEqual(true);
  });
});
