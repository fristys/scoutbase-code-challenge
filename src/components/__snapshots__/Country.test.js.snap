// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component Renders without crashing 1`] = `ShallowWrapper {}`;
